devito/core/cpu.py

```diff
@@ -240,9 +257,9 @@ def _make_clusters_passes_mapper(cls, **kwargs):
 
         # Callback used by `buffering`; it mimics `is_on_device`, which is used
         # on device backends
-        def callback(f):
+        def callback(f, *args):
             if f.is_TimeFunction and f.save is not None:
-                return f.time_dim
+                return f.space_dimensions
```

Comment:
for example? 
I don't think streaming n-dimensional volumes with n>3 is particularly clever performance wise...
I think we should wait until we have a compelling use case, run a systematic performance analysis, and take it from there
I can trivially change it as per your suggestion of course, but again, not sure whether it makes sense at this stage

Reply:
>n-dimensional volumes with n>3 
Not necessarily you could have some function `f[t, x, dim]` which or something. Not critical but would be safer to avoid space_dimension unless strictly necessary

Reply:
done